chunk_test.go

```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
 		}
 	})
 }
+
+func TestRetainChunkWOHighPrecision(t *testing.T) {
+	runDBTest(t, func(dbt *DBTest) {
+		query := "select 0"
+
+		ctx := WithArrowBatches(context.Background())
+
+		var rows driver.Rows
+		var err error
+
+		// must use conn.Raw so we can get back driver rows (an interface)
+		// which can be cast to snowflakeRows which exposes GetArrowBatch
+		err = dbt.conn.Raw(func(x interface{}) error {
+			queryer, implementsQueryContext := x.(driver.QueryerContext)
+			assertTrueF(t, implementsQueryContext, "snowflake connection driver does not implement queryerContext")
+
+			rows, err = queryer.QueryContext(WithArrowBatches(ctx), query, nil)
+			return err
+		})
+
+		assertNilF(t, err, "error running select query")
+
+		sfRows, isSfRows := rows.(SnowflakeRows)
+		assertTrueF(t, isSfRows, "rows should be snowflakeRows")
+
+		arrowBatches, err := sfRows.GetArrowBatches()
+		assertNilF(t, err, "error getting arrow batches")
+		assertEqualF(t, len(arrowBatches), 1, "should have one batch")
+
+		records, err := arrowBatches[0].Fetch()
+		assertNilF(t, err, "error getting batch")
+		assertNotNilF(t, records, "records should not be nil")
+
+		recs := *records
+		numRecords := len(recs)
+		assertEqualF(t, numRecords, 1, "should have exactly one record")
+
+		record := recs[0]
+		assertEqualF(t, len(record.Columns()), 1, "should have exactly one column")
+
+		column := record.Column(0)
```

Comment:
Maybe we can have `record.Column(0).(*array.Int8)`?

Reply:
good idea, updated